app/src/main/java/com/babylon/wallet/android/presentation/transaction/analysis/processor/TransactionTypeExtensions.kt

```diff
@@ -45,15 +47,21 @@ fun ExecutionSummary.involvedFungibleAddresses(excludeNewlyCreated: Boolean = tr
         }.toSet()
 }
 
-fun ExecutionSummary.involvedNonFungibleIds(excludeNewlyCreated: Boolean = true): Map<String, List<String>> {
+fun ExecutionSummary.involvedNonFungibleIds(excludeNewlyCreated: Boolean = true): Map<String, Set<String>> {
     val withdrawIndicators = accountWithdraws.values.flatten().filterIsInstance<ResourceIndicator.NonFungible>()
     val depositIndicators = accountDeposits.values.flatten().filterIsInstance<ResourceIndicator.NonFungible>()
-    return (withdrawIndicators + depositIndicators)
-        .filterNot {
-            excludeNewlyCreated && it.isNewlyCreated(this)
-        }.associate {
-            it.resourceAddress.addressString() to it.localIds
+    return (withdrawIndicators + depositIndicators).filterNot {
+        excludeNewlyCreated && it.isNewlyCreated(this)
+    }.fold(mutableMapOf(), operation = { acc, indicator ->
+        acc.apply {
+            if (containsKey(indicator.resourceAddress.addressString())) {
+                this[indicator.resourceAddress.addressString()] =
+                    this[indicator.resourceAddress.addressString()].orEmpty() + indicator.localIds
+            } else {
+                this[indicator.resourceAddress.addressString()] = indicator.localIds.toSet()
+            }
         }
+    })
```

Comment:
given that RET can return same non fungible address multiple times (depending on how manifest is constructed) this code was not properly returning all non fungible ids, but only the ones that were present in last resource indicator